source/extensions/filters/http/grpc_field_extraction/filter.cc

```diff
@@ -0,0 +1,229 @@
+#include "source/extensions/filters/http/grpc_field_extraction/filter.h"
+
+#include <memory>
+#include <string>
+
+#include "envoy/http/filter.h"
+#include "envoy/http/header_map.h"
+#include "envoy/registry/registry.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "source/common/buffer/zero_copy_input_stream_impl.h"
+#include "source/common/common/assert.h"
+#include "source/common/common/base64.h"
+#include "source/common/common/empty_string.h"
+#include "source/common/common/logger.h"
+#include "source/common/grpc/common.h"
+#include "source/common/grpc/status.h"
+#include "source/common/http/codes.h"
+#include "source/common/http/header_map_impl.h"
+#include "source/common/http/header_utility.h"
+#include "source/common/http/headers.h"
+#include "source/common/http/utility.h"
+#include "source/common/protobuf/protobuf.h"
+#include "source/extensions/filters/http/grpc_field_extraction/extractor.h"
+
+#include "absl/strings/escaping.h"
+#include "absl/strings/str_format.h"
+#include "proto_field_extraction/message_data/cord_message_data.h"
+#include "proto_field_extraction/message_data/message_data.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcFieldExtraction {
+namespace {
+using ::envoy::extensions::filters::http::grpc_field_extraction::v3::FieldExtractions;
+using ::envoy::extensions::filters::http::grpc_field_extraction::v3::GrpcFieldExtractionConfig;
+using ::Envoy::Grpc::Status;
+using ::Envoy::Grpc::Utility;
+
+// The filter prefixes.
+const char kRcDetailFilterGrpcFieldExtraction[] = "grpc_field_extraction";
+
+const char kRcDetailErrorRequestBufferConversion[] = "REQUEST_BUFFER_CONVERSION_FAIL";
+
+const char kRcDetailErrorTypeBadRequest[] = "BAD_REQUEST";
+
+const char kRcDetailErrorRequestFieldExtractionFailed[] = "REQUEST_FIELD_EXTRACTION_FAILED";
+
+const char kRcDetailErrorRequestOutOfData[] = "REQUEST_OUT_OF_DATA";
+
+const char kGrpcFieldExtractionDynamicMetadata[] = "envoy.filters.http.grpc_field_extraction";
```

Comment:
Can we consolidate this with `config.h` that also uses it?

Reply:
DONE